### PR TITLE
Support isProperty query in DataMirror.

### PR DESCRIPTION
**Pull request by mikeurbach**
### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Feature (or new API)

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

#### Release Notes

This adds an API to DataMirror to query if a Data is a Property.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Review by sequencer — APPROVED** [review]

**Review by azidar — APPROVED** [review]

**Comment by mikeurbach** [comment]
@sequencer I will plan to add the patches I'm submitting today to the 6.x release branch.

**Comment by mikeurbach** [comment]
Seems like there's a catch-22: if you add Please Merge label, mergify will merge main, but then easy cla will say the bot hasn't signed the agreement and block merging. I will merge this myself once CI passes.

**Comment by jackkoenig** [comment]
> Seems like there's a catch-22: if you add Please Merge label, mergify will merge main, but then easy cla will say the bot hasn't signed the agreement and block merging. I will merge this myself once CI passes.

I thought I had fixed this in https://github.com/chipsalliance/chisel/pull/3770, but it seems that only fixed it for backports. That being said, I'm not sure how useful Mergify automerging is considering Github has the automerge feature, maybe we should just remove that flow.

**Pull request merged**
